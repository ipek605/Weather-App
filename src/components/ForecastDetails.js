import React from "react";
import PropTypes from "prop-types";

function ForecastDetails(props) {
  const { forecast } = props;
  const { temperature, wind, humidity } = forecast;

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__temperature">
        <span className="forecast-details__temperature-max">
          {temperature.max}
        </span>
        <span className="forecast-details__temperature-min">
          {temperature.min}
          &deg;C
        </span>
      </div>
      <div className="forecast-details_wind-speed">
        {wind.speed}
        <span className="forecast-details_wind-direction">
          {wind.direction}
        </span>
      </div>
      <div className="forecast-details_humidity">
        {humidity}
        {}
      </div>
    </div>
  );
}
export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    humidity: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string
    }).isRequired
  }).isRequired
};
