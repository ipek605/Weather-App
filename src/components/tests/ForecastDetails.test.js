import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    temperature: {
      min: 12,
      max: 22
    },
    wind: {
      speed: 10,
      direction: "s"
    },
    humidity: 30
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastDetails instances", () => {
    const { getAllByTestId } = render(
      <ForecastDetails forecast={validProps} />
    );

    expect(getAllByTestId("forecast-details")).toHaveLength(1);
  });
});
