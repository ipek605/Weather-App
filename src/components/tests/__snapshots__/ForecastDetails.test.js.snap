// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__temperature"
    >
      <span
        class="forecast-details__temperature-max"
      >
        22
      </span>
      <span
        class="forecast-details__temperature-min"
      >
        12Â°C
      </span>
    </div>
    <div
      class="forecast-details_wind-speed"
    >
      10
      <span
        class="forecast-details_wind-direction"
      >
        s
      </span>
    </div>
    <div
      class="forecast-details_humidity"
    >
      30
    </div>
  </div>
</DocumentFragment>
`;
